### PR TITLE
add tutorial for creating registry, cache, and using it

### LINKED ISSUE #617: shpc remove to remove container.yaml recipes
Now that we will have a tutorial to create a registry, it would be helpful to have an shpc remove command, which would allow cloning and then emptying out a registry (to make your own). Currently you have to manually remove or loop through shpc show and do a custom remove command that also needs to account for the nesting. It's easier to provide:

```console
# remove all modules
$ shpc remove

# remove based on a pattern
$ shpc remove quay.io/biocontainers
```

### PR DESCRIPTION
**Pull request by vsoch**
This pull request will do the following:

- add a remove command to close #617 
- add a complete tutorial for creating a remote, a cache, and using it

It's not the easiest process, but I've worked for days fairly hard on it, and (at least beyond the registry generation) I think the actions are fairly straight forward. I'm not sure how to simplify this further, but definitely would be open to ideas (or more ideally some help!) :laughing: 

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

File: docs/getting_started/developer-guide.rst [hunk 1]
```diff
@@ -904,4 +916,252 @@ The reason we allow this additional listing is because the cache often misses be
...
+
+Remember that any container, once it goes into the registry, will have tags
+and digests automatically updated via the "Update Containers" action workflow.
+
+Creating a Cache
+----------------
+
+Instead of manually adding entries, let's create an automated way to populate
+entries from a cache. You can read more about the algorithm we use to derive aliases
+in the ``shpc-registry-cache <https://github.com/singularityhub/shpc-registry-cache>`_
```
[the remark this answers is not in the document]
Comment (georgiastuart) [reply docs/getting_started/developer-guide.rst:1043]: Or start with one backtick.

**Comment by georgiastuart** [comment]
Super cool and I was able to follow along easily!

File: docs/getting_started/developer-guide.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (vsoch) [reply docs/getting_started/developer-guide.rst:1043]: The latter - there should be just one (it's a typo) good catch!

Comment (vsoch) [inline docs/getting_started/developer-guide.rst:1043]:
```suggestion
in the `shpc-registry-cache <https://github.com/singularityhub/shpc-registry-cache>`_
```

**Comment by vsoch** [comment]
I'm not sure what is going on with installing podman, but it might be an issue with Circle - I'll try again later.

**Comment by marcodelapierre** [comment]
Super reviewers, super cool! So basically this seems ready for merging @vsoch ?

**Pull request merged**